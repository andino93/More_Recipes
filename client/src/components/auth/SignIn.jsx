import React, { Component } from 'react';
import PropTypes from 'react-proptypes';
import classnames from 'classnames';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { signInAction } from '../../actions/auth/signinAction';
import loginValidate from '../../utils/loginValidate';

/**
 *
 * @export
 * 
 * @class SignInComponent
 * 
 * @extends {React.Component}
 */
export class SignIn extends Component {
  /**
   * Creates an instance of SignInComponent.
   * 
   * @param {any} props
   * 
   * @memberof SignInComponent
   */
  constructor(props) {
    super(props);
    this.state = {
      loginDetails: {
        email: '',
        password: ''
      },
      errors: {}
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  /**
   * componentWillReceiveProps
   * @param {object }nextProps
   * @return {object} object
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.signInState.success === true) {
      $('.close').click();
    }
  }

  /**
  * Handle submit
  * 
  * @param {event} event
  * 
  * @return {event} event 
  * 
  */
  handleSubmit(event) {
    event.preventDefault();
    if (this.checkValidity()) {
      this.setState({ errors: {} });
      this.props.signInAction(this.state.loginDetails);
      this.setState({
        loginDetails: {
          email: '',
          password: ''
        }
      });
    }
  }
  /**
     * Check Validity
     * 
     * @param {null} event
     * 
     * @return {event} event

   */
  checkValidity() {
    const { errors, isValid } = loginValidate(this.state.loginDetails);
    if (!isValid) {
      this.setState({ errors });
    }
    return isValid;
  }

  /**
  * Handle change
  * 
  * @param {event} event
  * 
  * @return {event} event 
  */
  handleChange(event) {
    const loginDetails = this.state.loginDetails;
    loginDetails[event.target.name] = event.target.value;
    this.setState({ loginDetails });
  }
  /**
    * Renders signUp component
    * 
    * @returns {XML} XML/JSX
*/
  render() {
    const { loginDetails, errors } = this.state;
    const { signInState } = this.props;

    return (
      <div
        className="modal fade"
        id="signInModal"
        tabIndex="-1"
        role="dialog"
        aria-labelledby="signInModalLabel"
        aria-hidden="true">
        <div className="modal-dialog" role="document">
          <div className="modal-content simplebox blade">
            <div className="modal-header">
              <h5 className="modal-title" id="signInModalLabel">Sign In</h5>
              <button type="button" className="close" data-dismiss="modal"
                aria-label="Close">
                <span aria-hidden="true">&times;</span>
              </button>
            </div>
            <div className="modal-body">
              <form className="mt-15" onSubmit={this.handleSubmit}>
                { signInState.fails ?
                  <div id="error"
                    className="alert alert-danger alert-dismissible">
                    <button type="button" className="close" data-dismiss="alert"
                      aria-label="Close">
                      <span aria-hidden="true">x</span>
                    </button>
                    { signInState.fails }
                  </div> : null
                }
                <small className="form-text text-muted">
                  All input fields marked 
                  <span className="text-danger">*</span> are required
                </small>
                <div className="form-group">
                  <label htmlFor="email">
                    Email address <span className="text-danger">*</span>
                  </label>
                  <input
                    type="text"
                    className={
                      classnames('form-control',
                        { 'is-invalid': errors.email ? !!errors.email
                          : false }
                      )
                    }
                    name="email"
                    aria-describedby="email"
                    placeholder="Enter email"
                    onChange={this.handleChange}
                    value={loginDetails.email}
                  />
                  { errors.email ?
                    <span className="invalid-feedback">{ errors.email }</span>
                    : null
                  }
                  <small id="email" className="form-text text-muted">
                    We will never share your email with anyone else.
                  </small>
                </div>
                <div className="form-group">
                  <label htmlFor="password">
                    Password <span className="text-danger">*</span>
                  </label>
                  <input
                    type="password"
                    className={
                      classnames('form-control',
                        { 'is-invalid': errors.password ? !!errors.password
                          : false }
                      )
                    }
                    name="password"
                    placeholder="Password"
                    onChange={this.handleChange}
                    value={loginDetails.password}
                  />
                  { errors.password ?
                    <span className="invalid-feedback">
                      { errors.password }</span>
                    : null
                  }
                </div>
                <div className="modal-footer">
                  <button
                    type="submit"
                    id="submitSignIn"
                    className="btn btn-warning">
                    Log In
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
SignIn.propTypes = {
  signInState: PropTypes.object.isRequired,
  signInAction: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  signInState: state.signinReducer
});

const mapDispatchToProps = dispatch =>
  bindActionCreators({ signInAction }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(SignIn);