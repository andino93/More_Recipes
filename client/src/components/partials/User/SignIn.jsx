import React from 'react';
import ReactDOM from 'react-dom';

/**
 *
 *
 * @export
 * @class SignInComponent
 * @extends {React.Component}
 */
export default class SignInComponent extends React.Component {
    render() {
        return (
            <div
                className="modal fade"
                id="signInModal"
                tabIndex="-1"
                role="dialog"
                aria-labelledby="signInModalLabel"
                aria-hidden="true">
                <div className="modal-dialog" role="document">
                    <div className="modal-content simplebox blade">
                        <div className="modal-header">
                            <h5 className="modal-title" id="signInModalLabel">Sign In</h5>
                            <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                                <span aria-hidden="true">&times;</span>
                            </button>
                        </div>
                        <div className="modal-body">
                            <form className="mt-15">
                                <div className="form-group">
                                    <label htmlFor="exampleInputEmail1">Email address</label>
                                    <input
                                        type="email"
                                        className="form-control"
                                        id="exampleInputEmail1"
                                        aria-describedby="emailHelp"
                                        placeholder="Enter email"/>
                                    <small id="emailHelp" className="form-text text-muted">We'll never share your email with anyone else.</small>
                                </div>
                                <div className="form-group">
                                    <label htmlFor="exampleInputPassword1">Password</label>
                                    <input
                                        type="password"
                                        className="form-control"
                                        id="exampleInputPassword1"
                                        placeholder="Password"/>
                                </div>
                            </form>
                        </div>
                        <div className="modal-footer">
                            <a href="profile.html" className="btn btn-outline-warning" role="button">Submit</a>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}