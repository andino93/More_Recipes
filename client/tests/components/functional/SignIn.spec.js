import React from 'react';
import expect from 'expect';
import Adapter from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';
import mockStore from '../../__mocks__/mockStore';
import mountWrap from '../../__mocks__/utils';
import SignIn from '../../../src/components/auth/SignIn.jsx';

configure({ adapter: new Adapter() });

describe('<SignIn>', () => {
  const store = mockStore({ signinReducer: { signInState: {} } });
  const connectedSignIn = mountWrap(SignIn, store, {});
  it('should render SignIn component', (done) => {
    expect(connectedSignIn).toMatchSnapshot();
    done();
  });
  it('should check for componentWillReceiveProps', () => {
    connectedSignIn.find('SignIn').instance().componentWillReceiveProps({
      signInState: {
        success: '',
      },
    });
  });
  it('should simulate handleChange', (done) => {
    connectedSignIn.find('input').at(0).simulate('change');
    done();
  });
  it('should simulate handleSubmit', (done) => {
    connectedSignIn.find('form').at(0).simulate('submit', {
      preventDefault: () => {}
    });
    connectedSignIn.find('SignIn').instance().setState({
      loginDetails: {
        email: 'moh@gmail.com',
        password: 'somethiglight'
      }
    });
    connectedSignIn.find('form').at(0).simulate('submit', {
      preventDefault: () => {}
    });
    done();
  });
});