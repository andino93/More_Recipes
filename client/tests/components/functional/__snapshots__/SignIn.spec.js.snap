// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn> should render SignIn component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "blank",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(SignIn)>
        <SignIn
          signInAction={[Function]}
          signInState={
            Object {
              "signInState": Object {},
            }
          }
        >
          <div
            aria-hidden="true"
            aria-labelledby="signInModalLabel"
            className="modal fade"
            id="signInModal"
            role="dialog"
            tabIndex="-1"
          >
            <div
              className="modal-dialog"
              role="document"
            >
              <div
                className="modal-content simplebox blade"
              >
                <div
                  className="modal-header"
                >
                  <h5
                    className="modal-title"
                    id="signInModalLabel"
                  >
                    Sign In
                  </h5>
                  <button
                    aria-label="Close"
                    className="close"
                    data-dismiss="modal"
                    type="button"
                  >
                    <span
                      aria-hidden="true"
                    >
                      Ã—
                    </span>
                  </button>
                </div>
                <div
                  className="modal-body"
                >
                  <form
                    className="mt-15"
                    onSubmit={[Function]}
                  >
                    <small
                      className="form-text text-muted"
                    >
                      All input fields marked
                      <span
                        className="text-danger"
                      >
                        *
                      </span>
                       are required
                    </small>
                    <div
                      className="form-group"
                    >
                      <label
                        htmlFor="email"
                      >
                        Email address 
                        <span
                          className="text-danger"
                        >
                          *
                        </span>
                      </label>
                      <input
                        aria-describedby="email"
                        className="form-control"
                        name="email"
                        onChange={[Function]}
                        placeholder="Enter email"
                        type="text"
                        value=""
                      />
                      <small
                        className="form-text text-muted"
                        id="email"
                      >
                        We will never share your email with anyone else.
                      </small>
                    </div>
                    <div
                      className="form-group"
                    >
                      <label
                        htmlFor="password"
                      >
                        Password 
                        <span
                          className="text-danger"
                        >
                          *
                        </span>
                      </label>
                      <input
                        className="form-control"
                        name="password"
                        onChange={[Function]}
                        placeholder="Password"
                        type="password"
                        value=""
                      />
                    </div>
                    <div
                      className="modal-footer"
                    >
                      <button
                        className="btn btn-warning"
                        type="submit"
                      >
                        Log In
                      </button>
                    </div>
                  </form>
                </div>
              </div>
            </div>
          </div>
        </SignIn>
      </Connect(SignIn)>
    </Router>
  </BrowserRouter>
</Provider>
`;
